### PR TITLE
Update domain plot

### PR DESCRIPTION
**Pull request by kgaonkar6**
PR is related to #12 and #15 

- [x] add domain legend to the bottom ( internal to annoFuse ) 
- [x] add filtering inside the plot_breakpoint()
- [x] add Sample param to plot_breakpoint()
- [x] if non-coding genes in fusion add just gene and transcript structure 
- [x] For intergenic fusions, plots the breakpoint in the intergenic region closest to Gene1A or Gene1B
- [x] Removed "opt" from warning function

tested with KCNH1--AL590132.1,BRAF--ANTXR1 , LINC01019--LINC01019/IRX1 and KIAA1549--BRAF

In addition 
- [x] examples in plot_breakpoints and annoFuse_single_sample function calls 
- [x] vignette is updated with the new plot_breakpoint function call

### CODE REVIEW

**Comment by federicomarini** [comment]
Looks good to me!
We can even expand a little more on the examples, but for the time being I'd say it is safe to merge as it is - it could be risky in having conflicts with the other branch 😛

File: R/plot_breakpoints.R [hunk 1]
```diff
@@ -1,24 +1,72 @@
...
+#' right <- plot_breakpoints(sampleid = "BS_044XZ8ST", domainDataFrame = domainDataFrame, exons = exons, geneposition = "Right", fusionname = "ANTXR1--BRAF")
+#' ggpubr::ggarrange(left, right, align = "h")
+plot_breakpoints <- function(domainDataFrame = NULL,
                              exons = exons,
-                             geneposition = geneposition) {
+                             geneposition = NULL,
+                             sampleid = NULL,
+                             fusionname = NULL,
+                             leftBreakpoint = NULL,
+                             rightBreakpoint = NULL,
```
Comment (federicomarini) [inline R/plot_breakpoints.R:31]: I can work on the default values once it is merged with the `exemplary` branch
I'm gonna need your expertise there to clarify what some meaningful defaults would be (or if not default values, the ones we would suggest when we write the docs ;) )

File: R/annoFuse_single_sample.R [hunk 2]
```diff
@@ -74,7 +90,7 @@ annoFuse_single_sample <- function(fusionfileArriba = fusionfileArriba,
 
   # if StarFusion file is empty only run standardization for Arriba calls
   if (is_empty(STARFusioninputfile$FusionName) & !is_empty(Arribainputfile$"gene1--gene2")) {
-    warning(paste("No fusion calls in StarFusion : ", opt$fusionfileStarFusion))
+    warning(paste("No fusion calls in StarFusion "))
```
Comment (federicomarini) [inline R/annoFuse_single_sample.R:93]: Yup!

File: R/plot_breakpoints.R [hunk 3]
```diff
@@ -55,19 +104,20 @@ plot_breakpoints <- function(domainDataFrame = domainDataFrame,
...
       ylab("Coordinate") +
       ggtitle(paste0(geneposition, " fused Gene:", gene))
   }
 
   # add domain level information
-  if (any(!is.na(geneExons$DESC))) {
-    p1 <- p1 + geom_linerange(ymin = geneExons$domain_start, ymax = geneExons$domain_end, size = 6, aes(col = geneExons$DESC, x = gene)) + labs(color = "Domain")
+  if (!all(is.na(geneExons$DESC))) {
+    p1 <- p1 + geom_linerange(ymin = geneExons$domain_start, ymax = geneExons$domain_end, size = 6, aes(col = geneExons$DESC, x = gene)) + labs(color = "Domain") + theme(legend.position = "bottom") +
+      ggplot2::scale_colour_discrete(na.translate = F)
```
Comment (federicomarini) [inline R/plot_breakpoints.R:120]: nice! I like the legend much better now

**Review by federicomarini — APPROVED** [review]
All good in the hood!

**Comment by kgaonkar6** [comment]
Awesome thanks for reviewing!

**Comment by kgaonkar6** [comment]
@federicomarini I did have small to-dos to plot_recurrent_genes and plot_recurrent_fusion functions.
-[ ] bigger x axis labels ( internal to annoFuse )
-[ ] titles for plots ( internal to annoFuse )

let me know if you can add that to you existing branch or I can work on it after your `exemplary` branch gets merged.

**Comment by federicomarini** [comment]
I think it is easiest if you do the changes directly in `exemplary`.
I just merged master into that, so it has all your latest changes!

**Pull request merged**